src/ring/rack.clj

```diff
@@ -25,8 +25,9 @@
 
       def responsify(output)
         begin
-          return output[0].to_java if output.respond_to?(:size) && output.size > 1
-          output.each{|s| s.to_java}
+          retval = []
+          output.each {|s| retval.push(s)}
+          retval
```

Comment:
The thing is, from an Rack API perspective it is irrelevant what the exact type of the body is, i.e it is not always an array. If you do not call `each`, you get into a complete clusterfuck of "random" types in the clojure code. Always returning an array with strings makes everything simpler.